lib/eclipse/Parser/Parser.cpp

```diff
@@ -337,7 +340,13 @@ void ParserState::handleRandomText(const string_view& keywordString ) const {
     std::stringstream msg;
     std::string trimmedCopy = keywordString.string();
 
-    if (trimmedCopy == "/") {
+    if (lastSizeType == OTHER_KEYWORD_IN_DECK) {
+      errorKey = ParseContext::PARSE_EXTRA_RECORDS;
+      msg << "Too many records in keyword: " 
+          << lastKeyWord
+          << ".\n";
+    }
+    else if (trimmedCopy == "/") {
```

Comment:
I think the ordering should change here, i.e. first check the `trimmedCopy == "/"` condition, and then check the `OTHER_KEYWORD_IN_DECK` - that should also solve the failing tests - I think.